### PR TITLE
Feature/edit profile

### PR DESCRIPTION
**Pull request by Vladyslav-Shvorobuk**

### CODE REVIEW

**Review by PodstDO — DISMISSED** [review]

**Review by ElenaBorisenkoO — DISMISSED** [review]

File: src/app/pages/edit-profile/edit-profile.controller.js [hunk 1]
```diff
@@ -36,8 +35,7 @@ function EditProfileController($log, $scope, $rootScope, userProfileService, syn
     reader.readAsDataURL(file);
     reader.onloadend = function () {
       $scope.formInfo.ava = reader.result;
-      ava.src = reader.result;
-
+      $scope.$apply();
```
Comment (motifej) [inline src/app/pages/edit-profile/edit-profile.controller.js]: don't ever use  _$scope.$apply()_ !!!

File: src/app/pages/edit-profile/edit-profile.controller.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ function EditProfileController($log, $scope, $rootScope, userProfileService, syn
       if (file) {
         userProfileService.setProfileImage(file);
       }
+
       $scope.success = true;
       setTimeout(function () {
```
Comment (motifej) [inline src/app/pages/edit-profile/edit-profile.controller.js]: don't ever use any native methods !!!

**Review by motifej — CHANGES_REQUESTED** [review]

File: src/app/pages/edit-profile/edit-profile.controller.js (hunk 1 above)
Comment (Vladyslav-Shvorobuk) [reply src/app/pages/edit-profile/edit-profile.controller.js]: I changed it to $scope.$applyAsync().

File: src/app/pages/edit-profile/edit-profile.controller.js (hunk 2 above)
Comment (Vladyslav-Shvorobuk) [reply src/app/pages/edit-profile/edit-profile.controller.js]: I changed it to $timeout().

**Review by Arkant — DISMISSED** [review]

**Review by PodstDO — DISMISSED** [review]

**Review by Alexandra1003 — DISMISSED** [review]

File: src/app/pages/edit-profile/edit-profile.controller.js [hunk 3]
```diff
@@ -18,26 +18,25 @@ function EditProfileController($log, $scope, $rootScope, userProfileService, syn
       if (file) {
         userProfileService.setProfileImage(file);
       }
+
       $scope.success = true;
-      setTimeout(function () {
+      $timeout(function () {
         $state.go('profile');
       }, 1500);
```
Comment (motifej) [inline src/app/pages/edit-profile/edit-profile.controller.js]: why do you decide use this time for waiting?

Comment (Vladyslav-Shvorobuk) [reply src/app/pages/edit-profile/edit-profile.controller.js]: I based on my own experience, I tried to use another time, but this time seemed to me optimal.

Comment (Vladyslav-Shvorobuk) [reply src/app/pages/edit-profile/edit-profile.controller.js]: and this time will be enought for reading message about successful saving.

Comment (Vladyslav-Shvorobuk) [reply src/app/pages/edit-profile/edit-profile.controller.js]: I delete this $timeout.

**Review by PodstDO — DISMISSED** [review]

**Review by pseudopilot — DISMISSED** [review]

**Review by motifej — APPROVED** [review]
good Job!

**Review by PodstDO — APPROVED** [review]

**Review by Arkant — APPROVED** [review]

**Review by ElenaBorisenkoO — APPROVED** [review]

**Pull request merged**